src/user/views/adsManager/views/advanced/components/campaign/fields/BudgetField.tsx

```diff
@@ -69,17 +82,38 @@ export function BudgetField() {
           disabled={!isDraft && !advertiser.selfServiceSetPrice}
         />
 
-        {!advertiser.selfServiceSetPrice ? (
-          <Typography variant="body2">
-            {uiTextForCampaignFormat(values.format)} campaigns are priced at a
-            flat rate of{" "}
-            <strong>
-              ${values.price} {_.upperCase(values.billingType)}
-            </strong>
-            .
-          </Typography>
-        ) : (
-          <Stack direction="column" spacing={2} alignItems="flex-start">
+        <Stack direction="column" spacing={2} alignItems="flex-start">
+          <FormikRadioControl
+            name="billingType"
+            label="Billing Type"
+            onChange={(e) => {
+              const billingType = e.target.value.toUpperCase();
+              const price = props.prices.find(
+                (p) =>
+                  p.format === values.format && p.billingType === billingType,
+              );
+              if (price) campaignPrice.setValue(price?.billingModelPrice);
```

Comment:
nit: 

```suggestion
              if (price) campaignPrice.setValue(price.billingModelPrice);
```

You've qualified that price is defined, so you don't need to nullability check in the body of the if, typescript will infer this.